### PR TITLE
doc: refactor some code by creating some reducer for cart and wishlist

### PR DESCRIPTION
**Pull request by mariyasada**
-refactor context code by adding wishlist into the same reducer 
-adding loader to the product page.
-check the live preview

### CODE REVIEW

File: src/components/nav/Nav.jsx [hunk 1]
```diff
@@ -95,20 +110,29 @@ export const Nav = () => {
...
         ""
       )}
 
       {isloggedIn ? (
-        <span className="badge badge-cart flex-center text-size-sm text-white">
-          {getCartItemCount(cartItem)}
-        </span>
+        cartItem.length === 0 ? (
+          ""
+        ) : (
```
Comment (JantuDeb) [inline src/components/nav/Nav.jsx:127]: cartItem.length === 0 &&

File: src/context/utils/cartUtils.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export const itemExistInCart=(product,state)=>{
+    
 let isExist =false;
  state.cartItem.map((item)=>{
       if(item._id === product._id) {
```
Comment (JantuDeb) [inline src/context/utils/cartUtils.js]: `isExist = state.cartItem.some(item=>item._id===product._id)`

**Review by JantuDeb — APPROVED** [review]
This looks fine

File: src/context/Reducer/CartItemReducer.js [hunk 3]
```diff
@@ -2,21 +2,30 @@ export const CartItemReducer=(state,action)=>{
...
+                    return{...state, cartItem:action.payload}  
+                    
+          case "LOAD_WISHLIST_DATA":
+            return {...state,  wishListItem:action.payload};
+
+        case "ADD_ITEM_TO_WISHLIST":
+            return {...state,  wishListItem:action.payload,};
+
+        case "REMOVE_ITEM_FROM_WISHLIST":
+            return {...state,wishListItem:action.payload}
```
Comment (dev-enforced) [inline src/context/Reducer/CartItemReducer.js:28]: Can combine all these cases into a single one as:
```
          case "LOAD_WISHLIST_DATA":
          case "ADD_ITEM_TO_WISHLIST":
          case "REMOVE_ITEM_FROM_WISHLIST":
                    return {...state,wishListItem:action.payload}
```
Making the task to be achieved easier

File: src/pages/WishListPage/WishListPage.jsx [hunk 4]
```diff
@@ -1,22 +1,21 @@
 import React from "react";
 import { Link } from "react-router-dom";
-import {
-  ErrorMessage,
-  Productcard,
-  WishListCard,
-} from "../../components/index";
+import { Productcard, WishListCard } from "../../components/index";
```
Comment (dev-enforced) [inline src/pages/WishListPage/WishListPage.jsx:3]: Can remove the index part from here

File: src/components/Hamburger/Hamburger.jsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        </Link>
+        <Link to="/cart" className="hamburger-menu-item">
+          Cart
+        </Link>
+        <Link to="/wishlist" className="hamburger-menu-item">
+          Wishlist
+        </Link>
+        <Link to="/login" className="hamburger-menu-item">
+          Login
+        </Link>
```
Comment (dev-enforced) [inline src/components/Hamburger/Hamburger.jsx]: Can have a list of paths which you can map over
```
Can have a list of colors which you can map over

[c1,c2,c3,c4,c5],map((pathName,index)=>{
return (
<Link key={pathName} className=`hamburger-menu-item` to=`${pathName}`></Link>
)
})
```

**Review by dev-enforced — APPROVED** [review]
The live preview is working well as expected. Liked the use of toast messages which you have developed for this purpose along with custom loaders which you have made while loading the products and categories.Got confused with the btnDisable state variable but then got to realise its usage for avoiding multiple click on same item action side effect.
Do format your files using prettier on VSCode using its extension at the time of refactoring further. Helps format in a clean and understandable manner 
Good work @mariyasada !!

**Pull request merged**